### DISCUSSION TITLE
Is it possible to change a center of mass in MatterJS?

### DISCUSSION THREAD

**Discussion by ProgrammingLife** [photonstorm/phaser#6442] [Q&A] [answered]
Let's see we have some simple example of using matterjs like this one:
https://labs.phaser.io/edit.html?src=src\3.60\physics\matterjs\circular%20body.js

Trying to change the center of ball's mass: 
```
this.ball.centerOfMass.set( 100, 200 );
```
but nothing changes because as it noted here it's read only property: https://photonstorm.github.io/phaser3-docs/Phaser.Physics.Matter.Components.Mass.html

Somewhere over Internet I found some trick:
```js
        const offset = { x: 30, y: 0 };
        this.ball.body.position.x += offset.x;
        this.ball.body.position.y += offset.y;
        this.ball.body.positionPrev.x += offset.x;
        this.ball.body.positionPrev.y += offset.y;
```
But as you might see I works weirdly.

So is there a way to change the center of mass?

**Comment by photonstorm** [comment, marked as answer]
`this.matter.body.setCentre(body, centre, relative)` (see Matter docs for details).

**Reply by ProgrammingLife** [reply]
Looks it break a collider or at least debug lines so a collider goes through another bodies:
```js
this.matter.body.setCentre( this.ball.body, { x: 15, y: 15 }, true );
```
Is there a way to update physics to apply this new setting?

**Reply by ProgrammingLife** [reply]
Some good fix to a sprite but debug lines are still draw in a wrong place:
```js
const offset = { x: 15, y: 15 };
this.matter.body.setCentre( this.ball.body, offset, true );

const originX = this.ball.originX + ( offset.x / this.ball.displayWidth );
const originY = this.ball.originY + ( offset.y / this.ball.displayHeight );
this.ball.setOrigin( originX, originY );
```

**Reply by likeavenus** [reply]
@ProgrammingLife +++
@photonstorm Maybe need to create a container and add a sprite and a physical body in it to avoid the sprite falling out of the physical body?
<img width="161" alt="image" src="https://github.com/phaserjs/phaser/assets/37466798/5585abf6-d236-415e-bc02-c99eb336ff18">
